controller/server/gui.ml

```diff
@@ -64,7 +53,9 @@ let success msg =
 let error_handling =
   let open Opium_kernel.Rock in
   let filter handler req =
-    match%lwt handler req |> Lwt_result.catch with
+    (* Catch any exceptions that previously escaped Lwt *)
+    let res = try handler req with exn -> Lwt.fail exn in
```

Comment:
It’s fixing this manually added failure added in the info handler:

```ocaml
module InfoGui = struct
  let build app =
    app
    |> get "/info" (fun _ ->
        let _ = List.nth [] 0 in
        let%lwt server_info = Info.get () in
        page "info" [ "server_info", server_info |> Info.to_json ])
end
```

Nice! :+1:

Before the fix, It looks the same as the error you got when testing the proxy. The controller is still runnnig, but was failing in a request handler.

If I’m putting my failure code in the handler creation part, it’s making the controller to fail though:

```ocaml
module InfoGui = struct
  let build app =
    let _ = List.nth [] 0 in
    app
    |> get "/info" (fun _ ->
        let%lwt server_info = Info.get () in
        page "info" [ "server_info", server_info |> Info.to_json ])
end
```

Reply:
> If I’m putting my failure code in the handler creation part, it’s making the controller to fail though:

Yes, that's the scary bit. If it's an intermittent failure, systemd will be able restart it.

Reply:
There are also other potential points of failure, for example, an error in PlayOS.

One potential step forward, to reduce potential upgrade failures, could be to extract upgrade utility from the rest. It would complicate information on how the upgrade is going on though, from the controller.

Reply:
Yes, we could think about that at some point.